packages/machine/src/middleware/middleware.ts

```diff
@@ -156,12 +161,8 @@ export class NextMsgGenerator {
     internalMessage: InternalMessage,
     context: Context
   ) {
-    const res = getLastResult(Opcode.IO_WAIT, context.results);
-    // TODO: make getLastResult's return value nullable
-    // https://github.com/counterfactual/monorepo/issues/131
-    return JSON.stringify(res) === JSON.stringify({})
-      ? internalMessage.clientMessage
-      : res.value;
+    const res = context.intermediateResults.inbox;
+    return res === undefined ? internalMessage.clientMessage : res;
```

Comment:
@IIIIllllIIIIllllIIIIllllIIIIllllIIIIll I believe we can avoid the conditional expression by reducing it:

```ts
return res || internalMessage.clientMessage;
```